### ISSUE TITLE
Warn when comparing quoted values in `(( ))`

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#931]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
foo=1 bar=1
(( foo == "bar" ))

```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Warning: (( .. )) treats quoted string literals as variable names
